packages/ui-contract-editor/src/lib/ContractEditor/plugins/withClauses.js

```diff
@@ -75,12 +75,17 @@ const withClauses = (editor, withClausesProps) => {
   };
 
   editor.onChange = () => {
+    onChange();
     if (onClauseUpdated && editor.isInsideClause()) {
-      const debouncedOnClauseUpdated = _.debounce(onClauseUpdated, 1000, { maxWait: 10000 });
       const [clauseNode] = Editor.nodes(editor, { match: n => n.type === CLAUSE });
-      debouncedOnClauseUpdated(clauseNode[0]);
+      onClauseUpdated(clauseNode[0]).then(success => {
+        if (success) {
+          Transforms.setNodes(editor, { error: false }, { at: clauseNode[1] });
+        } else {
+          Transforms.setNodes(editor, { error: true }, { at: clauseNode[1] });
+        }
+      });
```

Comment:
Why are we removing debouncing here?

Reply:
Note I think this originated here: https://github.com/accordproject/cicero-ui/pull/189

Reply:
I think the parent app should instead decide if to debounce and for how long, so the user of the component has full control. I'll add it to the storybook demo though as an example 👍